cmd/scollector/collectors/fastly.go

```diff
@@ -50,8 +61,89 @@ const (
 	fastlyBillingBeforeDiscountDesc = "The total incurred cost plus extras cost this month."
 	fastlyBillingDiscountDesc       = "The calculated discount rate this month."
 	fastlyBillingCostDesc           = "The final amount to be paid this month."
+
+	fastlyStatusPrefix             = "fastly.status."
+	fastlyComponentStatusDesc      = "The current status of the %v. 0: Operational, 1: Degraded Performance, 2: Partial Outage, 3: Major Outage." // see iota for statusio.ComponentStatus
+	fastlyScheduledMaintDesc       = "The number of currently scheduled maintenances. Does not include maintenance that is current active"
+	fastlyActiveScheduledMaintDesc = "The number of currently scheduled maintenances currently in progress. Includes the 'in_progress' and 'verifying'"
+	fastlyActiveIncidentDesc       = "The number of currently active incidents. Includes the 'investingating', 'identified', and 'monitoring' states."
+)
+
+var (
+	fastlyStatusPopRegex = regexp.MustCompile(`(.*)\(([A-Z]{3})\)`)
 )
 
+func c_fastly_status(baseAddr string) (opentsdb.MultiDataPoint, error) {
+	var md opentsdb.MultiDataPoint
+	c := statusio.NewClient(baseAddr)
+	summary, err := c.GetSummary()
+	if err != nil {
+		return md, err
+	}
+
+	// Process Components (Pops, Support Systems)
+	for _, comp := range summary.Components {
+		match := fastlyStatusPopRegex.FindAllStringSubmatch(comp.Name, 1)
+		if len(match) != 0 && len(match[0]) == 3 { // We have a pop
+			//name := match[0][1]
+			code := match[0][2]
+			tagSet := opentsdb.TagSet{"code": code}
+			Add(&md, fastlyStatusPrefix+"pop", int(comp.Status), tagSet, metadata.Gauge, metadata.StatusCode, fmt.Sprintf(fastlyComponentStatusDesc, "pop"))
+			continue
+		}
+		// Must be service component
+		tagSet := opentsdb.TagSet{"service": comp.Name}
+		Add(&md, fastlyStatusPrefix+"service", int(comp.Status), tagSet, metadata.Gauge, metadata.StatusCode, fmt.Sprintf(fastlyComponentStatusDesc, "service"))
+	}
+
+	// Scheduled Maintenance
+	scheduledMaintByImpact := make(map[statusio.StatusIndicator]int)
+	activeScheduledMaintByImpact := make(map[statusio.StatusIndicator]int)
+	// Make Maps
+	for _, si := range statusio.StatusIndicatorValues {
+		scheduledMaintByImpact[si] = 0
+		activeScheduledMaintByImpact[si] = 0
+	}
+	// Group by scheduled vs inprogress/verifying
+	for _, maint := range summary.ScheduledMaintenances {
+		switch maint.Status {
+		case statusio.Scheduled:
+			scheduledMaintByImpact[maint.Impact]++
+		case statusio.InProgress, statusio.Verifying:
```

Comment:
I believe the unmarshal will err since it maps to the enum
On May 16, 2016 11:50 AM, "Craig Peterson" <notifications@github.com> wrote:

> In cmd/scollector/collectors/fastly.go
> <https://github.com/bosun-monitor/bosun/pull/1735#discussion_r63375879>:
>
> > +	}
> > +
> > +	// Scheduled Maintenance
> > +	scheduledMaintByImpact := make(map[statusio.StatusIndicator]int)
> > +	activeScheduledMaintByImpact := make(map[statusio.StatusIndicator]int)
> > +	// Make Maps
> > +	for _, si := range statusio.StatusIndicatorValues {
> > +		scheduledMaintByImpact[si] = 0
> > +		activeScheduledMaintByImpact[si] = 0
> > +	}
> > +	// Group by scheduled vs inprogress/verifying
> > +	for _, maint := range summary.ScheduledMaintenances {
> > +		switch maint.Status {
> > +		case statusio.Scheduled:
> > +			scheduledMaintByImpact[maint.Impact]++
> > +		case statusio.InProgress, statusio.Verifying:
>
> Is it worth logging in a default case? What happens if they add a new
> status?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/bosun-monitor/bosun/pull/1735/files/4140b74063d86880349236feeea6458422e69be5#r63375879>
>